### PR TITLE
[Bugfix] [Exam] StudentExam Detail View

### PR DESCRIPTION
**Pull request by anditurdiu**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context
Loads of:
`ERROR TypeError: Cannot read property '0' of undefined`
were thrown in the student exam detail view if the results were not present.

### Description
Add check if student participation is present before trying to access its properties

### Steps for Testing
1. Log in to Artemis
2. Navigate to an exam with registered users and generated exams, but no assessments/results
3. The view should load normally, list all exercises and no console errors should be shown

### Screenshots

image
image

### CODE REVIEW

**Review by stefanwaldhauser — APPROVED** [review]
Works as expected! Nice Work

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts [hunk 1]
```diff
@@ -96,19 +96,31 @@ export class StudentExamDetailComponent implements OnInit {
...
+     */
+    private setStudentScores() {
+        for (const exercise of this.studentExam?.exercises) {
             this.maxTotalScore += exercise.maxScore;
-            if (!!exercise.studentParticipations[0].results && exercise.studentParticipations[0].results.length >= 1) {
+            if (
+                !!exercise.studentParticipations &&
+                exercise.studentParticipations.length > 0 &&
+                !!exercise.studentParticipations[0].results &&
+                exercise.studentParticipations[0].results.length >= 1
```
Comment (JonasPetry) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts:118]:
```suggestion
                exercise.studentParticipations[0].results.length > 0
```

**Review by fde312 — APPROVED** [review]
Works like expected and code looks good too! Good job! 👍

**Review by MaisiKoleni — DISMISSED** [review]
Looks good to me and works (tested on TS3).

**Review by stefanwaldhauser — APPROVED** [review]
Reapprove

**Review by JonasPetry — APPROVED** [review]
Code looks good and works on TS3, good job!

File: src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts (hunk 1 above)
Comment (anditurdiu) [reply src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts]: 🗡️

**Review by kloessst — APPROVED** [review]
Works fine, code looks good

**Pull request merged**